### PR TITLE
tests: improves tests output by introducing subtests.

### PR DESCRIPTION
**Pull request by jcchavezs**
This PR ports some valid changes added in https://github.com/corazawaf/coraza/pull/298/files like `t.Run` calls or error extrapolation like using `%q` instead of `%s`.

> Thank you for contributing to Coraza WAF, your effort is greatly appreciated
> Before submitting check if what you want to add to `coraza` list meets [quality standards](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#quality-standards) before sending pull request. Thanks!

**Note**: _that go.mod and go.sum can only be modified for tested dependency updates or justified new features._

**Make sure that you've checked the boxes below before you submit PR:**

- [ ] My code includes positive and negative tests.
- [ ] I have an appropriate description with correct grammar.
- [ ] I have read [Contribution guidelines](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/github.com/corazawaf/coraza/v2sso/coraza-waf/blob/master/CONTRIBUTING.md#quality-standards).

Thanks for your PR :heart:

### CODE REVIEW

File: operators/ip_match_test.go [hunk 1]
```diff
@@ -45,15 +45,19 @@ func TestMultipleAddress(t *testing.T) {
 		t.Error("Cannot init ipmatchtest operator")
 	}
 	for _, ok := range addrok {
-		if !ipm.Evaluate(nil, ok) {
-			t.Errorf("Invalid result for single CIDR IpMatch " + ok)
-		}
+		t.Run(ok, func(t *testing.T) {
+			if !ipm.Evaluate(nil, ok) {
+				t.Errorf("Invalid result for single CIDR IpMatch " + ok)
```
Comment (jcchavezs) [inline operators/ip_match_test.go:50]:
```suggestion
				t.Errorf("Invalid result for single CIDR IpMatch %q", ok)
```

File: operators/ip_match_test.go [hunk 2]
```diff
@@ -45,15 +45,19 @@ func TestMultipleAddress(t *testing.T) {
...
+		})
 	}
 
 	for _, fail := range addrfail {
-		if ipm.Evaluate(nil, fail) {
-			t.Errorf("Invalid result for single CIDR IpMatch" + fail)
-		}
+		t.Run(fail, func(t *testing.T) {
+			if ipm.Evaluate(nil, fail) {
+				t.Errorf("Invalid result for single CIDR IpMatch" + fail)
```
Comment (jcchavezs) [inline operators/ip_match_test.go:58]:
```suggestion
				t.Errorf("Invalid result for single CIDR IpMatch %q", fail)
```

File: operators/operators_test.go [hunk 3]
```diff
@@ -37,71 +37,72 @@ type Test struct {
...
 				}
-			}
-			if strings.Contains(data.Param, `\x`) {
-				data.Param, err = strconv.Unquote(`"` + data.Param + `"`)
-				if err != nil {
-					t.Error("Cannot parse test case", err)
+				if strings.Contains(data.Param, `\x`) {
+					data.Param, err = strconv.Unquote(`"` + data.Param + `"`)
+					if err != nil {
+						t.Error("Cannot parse test case", err)
```
Comment (jcchavezs) [inline operators/operators_test.go:75]:
```suggestion
						t.Errorf("Cannot parse test case: %v", err)
```

File: operators/operators_test.go [hunk 4]
```diff
@@ -37,71 +37,72 @@ type Test struct {
...
+				data.Param = strings.ReplaceAll(data.Param, `\u0000`, "\u0000")
 
-			if strings.Contains(data.Input, `\x`) {
-				data.Input, err = strconv.Unquote(`"` + data.Input + `"`)
-				if err != nil {
-					t.Error("Cannot parse test case", err)
+				if strings.Contains(data.Input, `\x`) {
+					data.Input, err = strconv.Unquote(`"` + data.Input + `"`)
+					if err != nil {
+						t.Error("Cannot parse test case", err)
```
Comment (jcchavezs) [inline operators/operators_test.go:69]:
```suggestion
						t.Errorf("Cannot parse test case: %v", err)
```

File: operators/operators_test.go [hunk 5]
```diff
@@ -37,71 +37,72 @@ type Test struct {
...
-			if (res && data.Ret != 1) || (!res && data.Ret == 1) {
-				expected := "match"
-				if data.Ret == 0 {
-					expected = "no match"
+				if data.Name == "pmFromFile" || data.Name == "ipMatchFromFile" {
+					// read file
+					fname := root + "op/" + data.Param
+					d, err := os.ReadFile(fname)
+					if err != nil {
+						t.Errorf("Cannot open file %s", data.Param)
```
Comment (jcchavezs) [inline operators/operators_test.go:87]:
```suggestion
						t.Errorf("Cannot open file %q", data.Param)
```

File: operators/operators_test.go [hunk 6]
```diff
@@ -37,71 +37,72 @@ type Test struct {
...
+					}
+					data.Param = string(d)
+				}
+				if err := op.Init(data.Param); err != nil {
+					t.Error(err)
```
Comment (jcchavezs) [inline operators/operators_test.go:92]:
```suggestion
					t.Errorf("failed to init: %v", err)
```

File: operators/validate_nid_test.go [hunk 7]
```diff
@@ -31,15 +31,19 @@ func TestNidCl(t *testing.T) {
 	ok := []string{"11.111.111-1", "16100407-3", "8.492.655-8", "84926558", "111111111", "5348281-3", "10727393-k", "10727393-K"}
 	nok := []string{"11.111.111-k", "16100407-2", "8.492.655-7", "84926557", "111111112", "5348281-4"}
 	for _, o := range ok {
-		if !nidCl(o) {
-			t.Errorf("Invalid NID CL for %s", o)
-		}
+		t.Run(o, func(t *testing.T) {
+			if !nidCl(o) {
+				t.Errorf("Invalid NID CL for %s", o)
```
Comment (jcchavezs) [inline operators/validate_nid_test.go:36]:
```suggestion
				t.Errorf("Invalid NID CL for %q", o)
```

File: operators/validate_nid_test.go [hunk 8]
```diff
@@ -31,15 +31,19 @@ func TestNidCl(t *testing.T) {
...
+		})
 	}
 
 	for _, o := range nok {
-		if nidCl(o) {
-			t.Errorf("Valid NID CL for %s", o)
-		}
+		t.Run(o, func(t *testing.T) {
+			if nidCl(o) {
+				t.Errorf("Valid NID CL for %s", o)
```
Comment (jcchavezs) [inline operators/validate_nid_test.go:44]:
```suggestion
				t.Errorf("Valid NID CL for %q", o)
```

File: testing/coraza_test.go [hunk 9]
```diff
@@ -37,17 +37,22 @@ func TestEngine(t *testing.T) {
...
-				for _, mr := range test.transaction.MatchedRules {
-					debug += fmt.Sprintf(" %d", mr.Rule.ID)
-				}
-				t.Errorf("%s - %s: %s\nGot: %s\n%s\nREQUEST:\n%s", profile.Meta.Name, test.Name, e, debug, test.String(), test.Request())
+
+		t.Run(profile.Meta.Name, func(t *testing.T) {
+			for _, test := range tt {
+				t.Run(test.Name, func(t *testing.T) {
+					if err := test.RunPhases(); err != nil {
+						t.Error(err)
```
Comment (jcchavezs) [inline testing/coraza_test.go:45]:
```suggestion
						t.Errorf("failed to run phases: %v", err)
```

**Review by jptosso — APPROVED** [review]
LGTM

**Review by fzipi — APPROVED** [review]

File: operators/operators_test.go [hunk 10]
```diff
@@ -37,71 +37,72 @@ type Test struct {
...
-		err := json.Unmarshal(f, &cases)
-		if err != nil {
-			t.Error("Cannot parse test case", err)
-		}
-		for _, data := range cases {
-			// UNMARSHALL does not transform \u0000 to binary
-			data.Input = strings.ReplaceAll(data.Input, `\u0000`, "\u0000")
-			data.Param = strings.ReplaceAll(data.Param, `\u0000`, "\u0000")
+	for name, f := range files {
+		t.Run(name, func(t *testing.T) {
```
Comment (sts) [inline operators/operators_test.go:55]: probably want to remove tabs here

**Pull request merged**